### PR TITLE
adding tests for java.util.jar

### PR DESCRIPTION
**Pull request by maryqyh**

### CODE REVIEW

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarInputStreamTest.java [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+        while (count == 0 || entry != null) {
+            count++;
+            entry = jin.getNextEntry();
+        }
+        assertEquals(TOTAL_ENTRIES + 1, count);
+        jin.close();
+    }
+
+    /**
+     * These test the modified jar feature j2objc doesn't support,
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarInputStreamTest.java:173]: Add "J2ObjC removed" for easier future upgrade work?

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java [hunk 2]
```diff
@@ -0,0 +1,1067 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package libcore.java.util.jar;
+
+//import java.io.ByteArrayOutputStream;
```
Comment (lukhnos) [inline jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java:19]: Add `/* J2ObjC Removed... */` to make future upgrades easier? Same for other places in this file and in other files.

Comment (tomball) [reply jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java:19]: I'm not sure that's necessary if we instead move the "feature not supported on iOS" comment at the start of the class here.

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java [hunk 3]
```diff
@@ -0,0 +1,1067 @@
...
+//import java.util.zip.ZipException;
+//import java.util.zip.ZipFile;
+import junit.framework.TestCase;
+//import libcore.io.IoUtils;
+//import tests.support.resource.Support_Resources;
+//
+//
+public class JarFileTest extends TestCase {
+    /**
+     * These test the modified jar feature j2objc doesn't support,
```
Comment (tomball) [inline jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java:57]: Move this comment to below the copyright, so the rest of the file can be skipped by future developers.

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarInputStreamTest.java (hunk 1 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarInputStreamTest.java]: added the comment

File: jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java (hunk 2 above)
Comment (maryqyh) [reply jre_emul/android/platform/libcore/luni/src/test/java/libcore/java/util/jar/JarFileTest.java:24]: moved the comment to the start of class

**Pull request merged**